app/lib/pocketbase/CollectionMapping.ts

```diff
@@ -93,7 +93,12 @@ export const createCollectionMapping = <T extends ObjectWithId, Options extends
 				untrack(() => {
 					lists.set(listId, existingList ? { invalidated: false, ids: existingList?.ids } : undefined)
 					collection
-						.getFullList(options)
+						.getFullList({
+							...options,
+							// Use a unique request key based on the filter to prevent cancellation
+							// of requests for different records
+							requestKey: options?.filter ? `${name}-${options.filter}` : undefined
```

Comment:
Can you describe when and how this problem occurs? I don't have problem with request cancellations. This feels like a dirty workaround to some problem that should be addressed directly.

Reply:
When you add the same block to the page two or more times, the subsequent requests get cancelled, so all but the first block show up empty.

Reply:
Oh, right. I see that default requestKey is set to HTTP_METHOD + path, which is not what we want. Requests to the same collection get auto-cancelled. We might want to use listId as the requestKey.

Source: https://github.com/pocketbase/js-sdk/blob/master/README.md#auto-cancellation

```suggestion
							// Use a unique request key based on the list ID to prevent cancellation
							// of requests for different lists
							requestKey: listId
```